api/src/test/java/org/openmrs/api/db/hibernate/HibernateDiagnosisDAOTest.java

```diff
@@ -0,0 +1,137 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api.db.hibernate;
+
+import java.util.Date;
+import java.util.List;
+
+import org.junit.Assert;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.Before;
+import org.junit.rules.ExpectedException;
+import org.openmrs.CodedOrFreeText;
+import org.openmrs.Condition;
+import org.openmrs.Concept;
+import org.openmrs.ConceptName;
+import org.openmrs.ConditionVerificationStatus;
+import org.openmrs.Diagnosis;
+import org.openmrs.Encounter;
+import org.openmrs.Patient;
+import org.openmrs.User;
+import org.openmrs.api.db.DiagnosisDAO;
+import org.openmrs.test.BaseContextSensitiveTest;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class HibernateDiagnosisDAOTest extends BaseContextSensitiveTest {
+
+	private static final String DIAGNOSIS_XML = "org/openmrs/api/include/HibernateDiagnosisDAOTestDataset.xml";
+
+	@Autowired
+	DiagnosisDAO diagnosisDAO;
+
+	@Before
+	public void setUp() {
+		executeDataSet(DIAGNOSIS_XML);
+
+	}
+
+	@Test
+	public void shouldSaveDiagnosis() {
+		CodedOrFreeText codedOrFreeText = new CodedOrFreeText(new Concept(4),
+			new ConceptName(5089), "non coded");
+		Date date = new Date();
+		int id = 5;
+		String uuid = "4e663d96-6b78-11e0-93c3-18a9b5e044dc";
+		int rank = 2;
+		ConditionVerificationStatus certainty = ConditionVerificationStatus.CONFIRMED;
+		User creator = new User(1);
```

Comment:
If you are using these variables in one place, can you avoid declaring them?
For instance, you could just use diagnosis.setEncounter(new Encounter(3));